tests/utils.test.ts

```diff
@@ -0,0 +1,124 @@
+import { AppName } from "@/models/AppName";
+import {
+  filterFutureAnnouncements,
+  sortAnnouncementsByStartDate,
+  calculateTimeRemaining,
+  getEarliestAnnouncements,
+} from "../src/utils/utils";
+import { Announcement } from "@/models/Announcement";
+
+const announcements: Announcement[] = [
+  {
+    id: "1",
+    apps: [AppName.EATERY],
+    body: "Announcement 1",
+    endDate: new Date("2024-04-01T00:00:00Z"),
+    imageUrl: "image1.jpg",
+    link: "link1",
+    startDate: new Date("2024-03-15T00:00:00Z"),
+    title: "Announcement 1",
+  },
+  {
+    id: "2",
+    apps: [AppName.RESELL, AppName.COURSEGRAB],
+    body: "Announcement 2",
+    endDate: new Date("2024-08-20T00:00:00Z"),
+    imageUrl: "image2.jpg",
+    link: "link2",
+    startDate: new Date("2024-08-10T00:00:00Z"),
+    title: "Announcement 2",
+  },
+  {
+    id: "3",
+    apps: [AppName.UPLIFT],
+    body: "Announcement 3",
+    endDate: new Date("2025-03-30T00:00:00Z"),
+    imageUrl: "image3.jpg",
+    link: "link3",
+    startDate: new Date("2025-03-25T00:00:00Z"),
+    title: "Announcement 3",
+  },
+];
+
+const duplicateStartAnnouncements: Announcement[] = [
+  {
+    id: "4",
+    apps: [AppName.EATERY],
+    body: "Announcement 4",
+    endDate: new Date("2024-05-01T00:00:00Z"),
+    imageUrl: "image4.jpg",
+    link: "link4",
+    startDate: new Date("2024-03-15T00:00:00Z"),
+    title: "Announcement 4",
+  },
+  ...announcements,
+];
+
+describe("Utils", () => {
+  describe("filterFutureAnnouncements", () => {
+    it("should filter announcements to only include those with a start date in the future", () => {
+      const result = filterFutureAnnouncements(announcements);
+      expect(result.length).toBe(2);
+      expect(result[0].id).toBe("2");
+      expect(result[1].id).toBe("3");
+    });
+  });
+
+  describe("sortAnnouncementsByStartDate", () => {
+    it("should sort announcements by their start date in ascending order", () => {
+      const result = sortAnnouncementsByStartDate(announcements);
+      expect(result[0].id).toBe("1");
+      expect(result[1].id).toBe("2");
+      expect(result[2].id).toBe("3");
+    });
+  });
+
+  describe("calculateTimeRemaining", () => {
+    it("should calculate the time remaining until the given start date", () => {
```

Comment:
Can you break up this test to calculate remaining time for days, hours, minutes, and seconds, precisely? Just because `days` works doesn't mean the others will.